docs/DeveloperGuide.adoc

```diff
@@ -137,76 +146,174 @@ image:BetterModelClassDiagram.png[]
+
+Step 2: The RequirementAddCommand is executed to add the new requirement to the model. In this step, the following check is performed:
+
+* Check if a requirement with the same name already exists in the course book.
+
+Step 3: The new requirement is added to the course book.
+
+
 === [Proposed] Undo/Redo feature
+
 ==== Proposed Implementation
 
-The undo/redo mechanism is facilitated by `VersionedAddressBook`.
-It extends `AddressBook` with an undo/redo history, stored internally as an `addressBookStateList` and `currentStatePointer`.
+The undo/redo mechanism is facilitated by `VersionedCourseBook`.
+It extends `CourseBook` with an undo/redo history, stored internally as an `addressBookStateList` and `currentStatePointer`.
 Additionally, it implements the following operations:
 
-* `VersionedAddressBook#commit()` -- Saves the current address book state in its history.
-* `VersionedAddressBook#undo()` -- Restores the previous address book state from its history.
-* `VersionedAddressBook#redo()` -- Restores a previously undone address book state from its history.
+* `VersionedCourseBook#commit()` -- Saves the current course book state in its history.
+* `VersionedCourseBook#undo()` -- Restores the previous course book state from its history.
+* `VersionedCourseBook#redo()` -- Restores a previously undone course book state from its history.
 
-These operations are exposed in the `Model` interface as `Model#commitAddressBook()`, `Model#undoAddressBook()` and `Model#redoAddressBook()` respectively.
+These operations are exposed in the `Model` interface as `Model#commitCourseBook()`, `Model#undoCourseBook()` and `Model#redoCourseBook()` respectively.
 
 Given below is an example usage scenario and how the undo/redo mechanism behaves at each step.
 
-Step 1. The user launches the application for the first time. The `VersionedAddressBook` will be initialized with the initial address book state, and the `currentStatePointer` pointing to that single address book state.
+Step 1. The user launches the application for the first time.
+The `VersionedCourseBook` will be initialized with the initial course book state, and the `currentStatePointer` pointing to that single memo book state.
 
 image::UndoRedoState0.png[]
 
-Step 2. The user executes `delete 5` command to delete the 5th person in the address book. The `delete` command calls `Model#commitAddressBook()`, causing the modified state of the address book after the `delete 5` command executes to be saved in the `addressBookStateList`, and the `currentStatePointer` is shifted to the newly inserted address book state.
+<<<<<<< HEAD Step 2. The user executes `delete 5` command to delete the 5th module in the course book.
+The `delete` command calls `Model#commitCourseBook()`, causing the modified state of the course book after the `delete 5` command executes to be saved in the `addressBookStateList`, and the `currentStatePointer` is shifted to the newly inserted course book state.
```

Comment:
3.3 to 3.3.2: information was updated to be relevant to product but diagrams for this section was not updated to reflect the product.